server/src/main/java/org/opensearch/search/backpressure/stats/SearchShardTaskStats.java

```diff
@@ -30,21 +31,27 @@
 public class SearchShardTaskStats implements ToXContentObject, Writeable {
     private final long cancellationCount;
     private final long limitReachedCount;
+    private Long completionCount;
```

Comment:
This should be declared `final`.

Reply:
I did not declare it final because it would force me to add the `else` clause in the `StreamInput` constructor even when we are using `readOptionalVLong()`.
second since this member variable is private and there are no setters, this will only be initialised once i,e; in the constructor. 

Do you see any other value add of final here ?

Reply:
Explicitly assigning it to `null` in an `else` block is perfectly fine and preferable. These stats' classes are merely data carriers (i.e. [records](https://www.baeldung.com/java-record-keyword#purpose)) so the fields aren't expected to change (no need for setters/modifiers).